lib/profile.go

```diff
@@ -20,46 +19,82 @@ import (
 // changes. all of this code should be refactored into subroutines of general
 // configuration getters & setters
 type ProfileMethods struct {
-	inst *Instance
+	d dispatcher
 }
 
-// CoreRequestsName implements the Request interface
-func (ProfileMethods) CoreRequestsName() string { return "profile" }
+// Name returns the name of this method group
+func (m ProfileMethods) Name() string {
+	return "profile"
+}
 
-// NewProfileMethods creates a ProfileMethods pointer from either a repo
-// or an rpc.Client
-func NewProfileMethods(inst *Instance) *ProfileMethods {
-	return &ProfileMethods{inst: inst}
+// Attributes defines attributes for each method
+func (m ProfileMethods) Attributes() map[string]AttributeSet {
+	return map[string]AttributeSet{
+		"getprofile":      {denyRPC, ""},
+		"setprofile":      {denyRPC, ""},
+		"setprofilephoto": {denyRPC, ""},
+		"setposterphoto":  {denyRPC, ""},
+	}
 }
 
+type ProfileParams struct{}
+
 // GetProfile get's this node's peer profile
-func (m *ProfileMethods) GetProfile(ctx context.Context, in *bool) (*config.ProfilePod, error) {
-	res := &config.ProfilePod{}
-	var err error
+func (m ProfileMethods) GetProfile(ctx context.Context, p *ProfileParams) (*config.ProfilePod, error) {
+	got, _, err := m.d.Dispatch(ctx, dispatchMethodName(m, "getprofile"), p)
+	if res, ok := got.(*config.ProfilePod); ok {
+		return res, err
+	}
+	return nil, dispatchReturnError(got, err)
+}
 
-	if m.inst.http != nil {
-		err = m.inst.http.CallMethod(ctx, AEProfile, http.MethodGet, nil, res)
-		if err != nil {
-			return nil, err
-		}
-		return res, nil
+// SetProfileParams defines parameters for setting parts of a profile
+// Cannot use this to set private keys, your peername, or peer id
+type SetProfileParams struct {
+	Pro *config.ProfilePod
+}
+
+// SetProfile stores changes to the active peer's editable profile
+func (m ProfileMethods) SetProfile(ctx context.Context, p *SetProfileParams) (*config.ProfilePod, error) {
+	got, _, err := m.d.Dispatch(ctx, dispatchMethodName(m, "setprofile"), p)
+	if res, ok := got.(*config.ProfilePod); ok {
+		return res, err
 	}
+	return nil, dispatchReturnError(got, err)
+}
 
-	var pro *profile.Profile
-	r := m.inst.repo
+// FileParams defines parameters for Files as arguments to lib methods
+type FileParams struct {
+	// Url      string    // url to download data from. either Url or Data is required
+	Filename string    // filename of data file. extension is used for filetype detection
```

Comment:
This needs the struct tag `qri:"fspath"`. Also, comment should move up above the field instead of to the right.